### PR TITLE
Microphone live inference catching up when inference is too slow (whisper).

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

When using relatively slow inference models (like whisper, especially the large variants
on moderate hardware). The live inference snippets would be so slow that it would
feel extremely laggy.

This PR fixes it by adding an estimate of what is real time, and simply skip inferences
when we're too far behind.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/pipelines/audio_utils.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import datetime
```
Comment (sgugger) [inline src/transformers/pipelines/audio_utils.py:1]: This file should have a Copyright btw.

**Review by sgugger — APPROVED** [review]
Thanks for the fix!

**Review by ArthurZucker — APPROVED** [review]
Thanks! LGTM

File: src/transformers/pipelines/audio_utils.py (hunk 1 above)
Comment (Narsil) [reply src/transformers/pipelines/audio_utils.py:2]: Added.

**Pull request merged**